File: Two Pointers/Easy/LET_no922_Sort Array By Parity II.cpp
class Solution {
public:
    vector<int> sortArrayByParityII(vector<int>& nums) {
        for (int i = 0, j = 1; i < nums.size() && j < nums.size(); ) {
            if (nums[i] % 2 == 0) i += 2;
            else if (nums[j] % 2 == 1) j += 2;
            else swap(nums[i], nums[j]);
        }
        return nums;
    }
};

File: Two Pointers/Easy/LET_no922_Sort Array By Parity II.py
class Solution:
    def sortArrayByParityII(self, nums: List[int]) -> List[int]:
        i, j, n = 0, 1, len(nums)
        while i < n and j < n:
            if nums[i] % 2 == 0: i += 2
            elif nums[j] % 2 == 1: j += 2
            else: nums[i], nums[j] = nums[j], nums[i]
        
        return nums